File: frontend/src/components/pages/onboarding/api/hooks/definitions/expense-payment-schedule-definition.ts
import { ExpensePaymentTiming } from 'components/pages/onboarding/enums/expense-payment-timing';
import { PaycheckPosition } from 'components/pages/onboarding/enums/paycheck-position';

export default interface ExpensePaymentScheduleDefinition {
  source_key: string;
  timing: ExpensePaymentTiming;
  paycheck_position: PaycheckPosition | null;
  day_of_month: number | null;
  auto_deducted: boolean;
}

File: backend/debt_profile/types/expense_payment_schedule_definition.py
from typing import TypedDict

from debt_profile.models import ExpensePaymentTiming, PaycheckPosition


class ExpensePaymentScheduleDefinition(TypedDict):
    source_key: str
    timing: ExpensePaymentTiming
    paycheck_position: PaycheckPosition | None
    day_of_month: int | None
    auto_deducted: bool
